HabitTracker.Jackua/Controller/HabitTrackerController.cs

```diff
@@ -0,0 +1,542 @@
+                        {
+                            HabitId = reader.GetInt32(0),
+                            HabitName = reader.GetString(1)
+                        });
+                }
+            }
+            else
+            {
+                MenuView.NoRows();
+            }
+
+            connection.Close();
+
+            MenuView.DashLines();
+            foreach (var hm in tableData)
+            {
+                MenuView.DisplayData(hm);
+            }
+            MenuView.DashLines();
+        }
+    }
+
+    private static void GetSummary()
+    {
+        Console.Clear();
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+
+            var tableCmd = connection.CreateCommand();
+            tableCmd.CommandText =
+                @" 
+                SELECT (
+                        SELECT habitname 
+                        FROM habit 
+                        where recordHabit = habitId
+                       ), 
+                    SUM(quantity) 
+                    FROM record 
+                    GROUP BY recordHabit
+                ;";
+
+            Dictionary<string, int> tableData = new();
+
+            SqliteDataReader reader = tableCmd.ExecuteReader();
+
+            if (reader.HasRows)
+            {
+                while (reader.Read())
+                {
+                    tableData.Add(
+                        reader.GetString(0),
+                        reader.GetInt32(1)
+                    );
+                }
+            }
+            else
+            {
+                MenuView.NoRows();
+            }
+
+            connection.Close();
+
+            MenuView.DashLines();
+            foreach (string key in tableData.Keys)
+            {
+                MenuView.DisplayData(key, tableData[key]);
+            }
+            MenuView.DashLines();
+        }
+    }
+
+    private static void InsertRecord()
+    {
+        Console.Clear();
+        
+        GetAllHabits();
+        MenuView.HabitIdRequest();
+        int habitId = GetNumberInput();
+        if (habitId == 0) return;
+
+        using (var connection = new SqliteConnection(connectionString))
+        {
+            connection.Open();
+
+            var checkCmd = connection.CreateCommand();
+
+            checkCmd.CommandText = $"SELECT EXISTS(SELECT 1 FROM habit WHERE habitId = {habitId})";
+            int checkQuery = Convert.ToInt32(checkCmd.ExecuteScalar());
+
+            if (checkQuery == 0)
+            {
+                MenuView.DoesNotExist(habitId, "Habit");
+                connection.Close();
+                return;
+            }
+
+            string date = GetDateInput();
+            if (date == "0") return;
+
+            MenuView.QuantityRequest();
+            int quantity = GetNumberInput();
+            if (quantity == 0) return;
+
+            var tableCmd = connection.CreateCommand();
+
+            tableCmd.CommandText =
```

Comment:
This code is open to [SQL Injection Attacks](https://cheatsheetseries.owasp.org/cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.html)

Remember to parameterise your queries:
```
// Open to SQL Injection Attacks.
command.CommandText = $"INSERT INTO Table(Column) VALUES ('{value}');";

// vs

// Closed to SQL Injection Attacks.
command.CommandText = $"INSERT INTO Table(Column) VALUES (@Value);";
command.Parameters.AddWithValue("@Value", value);
```